pretyping/typing.ml

```diff
@@ -49,18 +49,20 @@ let e_assumption_of_judgment env evdref j =
   with Type_errors.TypeError _ | PretypeError _ ->
     error_assumption env !evdref j
 
-let e_judge_of_apply env evdref funj argjv =
+let e_judge_of_applied_inductive_knowing_parameters env evdref funj ind argjv =
   let rec apply_rec n typ = function
   | [] ->
       { uj_val  = mkApp (j_val funj, Array.map j_val argjv);
-        uj_type = typ }
-  | hj::restjl ->
-      match EConstr.kind !evdref (whd_all env !evdref typ) with
+        uj_type = whd_all env !evdref (EConstr.of_constr (snd (Term.decompose_prod_n_assum n (inductive_type_knowing_parameters env !evdref ind argjv)))) }
```

Comment:
This seems suspicious, we reduce in the original `env` even though we are under binders.

Reply:
Indeed, thanks. I fixed it.